src/duplicate.coffee

```diff
@@ -1,7 +1,181 @@
 # Description:
-#   A hubot script to do nothing
+#   A hubot script to duplicate messages between services
+#
+# Configuration:
+#   HUBOT_DUPLICATE_TO
+#   HUBOT_DUPLICATE_FROM
+#   HUBOT_DUPLICATE_ROOMS
+#
+# Commands:
+#   hubot duplicate about - Echoes an expanded about message
 #
 # Author:
 #   Andrew Lucas (sqweelygig) <andrewl@resin.io> <sqweelygig@gmail.com>
+class Duplicator
+	constructor: (robot) ->
+		@to =
+			name: process.env.HUBOT_DUPLICATE_TO
+			adapter: require('hubot-' + process.env.HUBOT_DUPLICATE_TO).use(robot)
+		@from = { name: process.env.HUBOT_DUPLICATE_FROM }
+		@pairs = JSON.parse process.env.HUBOT_DUPLICATE_ROOMS
+		@apiKeys = JSON.parse(robot.brain.get('ScriptDuplicateAPIKeys')) ? {}
+		@persists = new Set(Object.keys @apiKeys)
+		@robot = robot
+		@knownThreads = {}
+		@setupActions robot
+
+	setupActions: (robot) ->
+		# Core function
+		robot.hear /.*/i, (context) =>
+			@echo(
+				context.message.text
+				{
+					user: context.message.user.name
+					flow: context.message.room
+					comment: context.message.id
+					thread: context.message.metadata.thread_id
+				}
+			)
+		# My key
+		robot.respond /set my key (\w+)/i, (context) => @setAPIKey context.message.user.name, context.match[1], @respond(context)
+		robot.respond /check my key (\w+)/i, (context) => @checkAPIKey context.message.user.name, context.match[1], @respond(context)
+		# My regex
+		robot.respond /set my pattern (\w+)/i, (context) => @setPattern context.message.user.name, context.match[1], @respond(context)
+		robot.respond /reveal my pattern/i, (context) => @viewPattern context.message.user.name, @respond(context)
+		# My details
+		robot.respond /save my config(uration)?/i, (context) => @persistUser context.message.user.name, @respond(context)
+		robot.respond /forget my config(uration)?/i, (context) => @forgetUser context.message.user.name, @respond(context)
+		# Other's details
+		robot.respond /reveal who.*s.? config(ured)?/i, (context) => @listUsers @respond(context)
+		robot.respond /forget @(\w+)'?s? config(uration)?/i, (context) => @forgetUser context.match[1], @respond(context)
+		# Misc
+		robot.respond /duplicator( about| help)?/i, (context) => @viewHelp @respond(context)
+		robot.respond /reveal your config(uration)?/i, (context) => @viewEnvVars @respond(context)
+
+	# text, { user: [id], flow: [id], comment: [id], thread: [id] }
+	# Duplicate this message to the paired room with the paired identity
+	echo: (text, from_ids) ->
+		to_ids = {
+			flow: null
+			user: null
+			comment: null
+			thread: null
+		}
+		to_ids.thread = @knownThreads[from_ids.thread]
+		for pair in @pairs when pair[@from.name] is from_ids.flow
+			to_ids.flow = pair[@to.name]
+		for username, object of @apiKeys when from_ids.user.match new RegExp(object.pattern ? username)
```

Comment:
object is not a super helpful name :smile:. You could destructure this very nicely with:

```coffeescript
for username, { key, pattern } of @apiKeys when from_ids.user.match(new RegExp(pattern ? username))
  to_ids.user = key
```